assessment/models.py

```diff
@@ -576,6 +618,41 @@ def create_assignment_attempt(self, assignment: Assignment) -> AssignmentAttempt
         )
         return assignment_attempt
 
+    def get_all_assessment_tool_attempts(self):
+        return self.attempt.toolattempt_set.all()
```

Comment:
Hello, Jo. Allow me to ask, if it were to be the case that self.attempt is None or null, will this code throw an error? I believe that if the error is this deep its unlikely to be caught by the try except blocks outside.

Reply:
Hello, Alex. According to the code that handles the assessee registration, an instance of TestFlowAttempt will always be created and assigned to the AssessmentEventParticipation. Hence, there should not be any case where the self.attempt is Null. 

Regards,
Johanes Christian